shim.go

```diff
@@ -96,6 +98,36 @@ func newShimConfig(config PodConfig) interface{} {
 	}
 }
 
+func shimRunning(pid int) (bool, error) {
+	process, err := os.FindProcess(pid)
+	if err != nil {
+		return false, err
+	}
+
+	if err := process.Signal(syscall.Signal(0)); err != nil {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+func stopShim(pid int) error {
+	running, err := shimRunning(pid)
```

Comment:
This branch is effectively doing:

```
kill(pid, 0)
<< check for error denoting ESRCH >>
kill(pid, SIGKILL)
```
There is no point doing two kills and its racy since the process might exist when you `kill(pid, 0)`, then die before you `kill(pid, SIGKILL)` it. All you need to do is the actual `SIGKILL` and if the process doesn't exist at that point, you'll get a `ESRCH` back which you can check for.